### PR TITLE
xwayland: Use mask flag to configure appropriate geometry elements

### LINKED ISSUE #309: Some xwayland windows start with wrong size
xterm can end up at 3x3 instead of the expected size, with decoration plugin enabled.

### PR DESCRIPTION
**Pull request by soreau**
Pass mask to wlr_xwayland_surface_configure() so as to not configure
a window twice with the same size. If this happens between commits,
the client might not end up with its intended size. This requires
[wlroots #1843](https://github.com/swaywm/wlroots/pull/1843). Fixes #309.

### CODE REVIEW

**Comment by soreau** [comment]
I see what you mean. Is there a case you have in mind that might exhibit a problem?

**Comment by ammen99** [comment]
Yes, if a window is moved, the client needs to be notified because it needs to reposition its menus (VLC is a good test client, with the X11 backend)

**Comment by soreau** [comment]
Ok thanks, I'll look into it.

File: src/view/xwayland.cpp [hunk 1]
```diff
@@ -178,7 +178,7 @@ class wayfire_xwayland_view_base : public wf::wlr_view_t
                 &output_geometry_changed);
         }
         /* Update the real position */
-        send_configure();
+        send_configure(XCB_CONFIG_WINDOW_X | XCB_CONFIG_WINDOW_Y | XCB_CONFIG_WINDOW_WIDTH | XCB_CONFIG_WINDOW_HEIGHT);
```
Comment (ammen99) [inline src/view/xwayland.cpp]: why do we need width and height here? It is again position-update-only

File: src/view/xwayland.cpp [hunk 2]
```diff
@@ -344,7 +343,11 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
 
         last_server_width = w;
         last_server_height = h;
-        send_configure(w, h);
+
+        if (w != geometry.width && h != geometry.height)
+            send_configure(w, h, XCB_CONFIG_WINDOW_X | XCB_CONFIG_WINDOW_Y |
+                XCB_CONFIG_WINDOW_WIDTH | XCB_CONFIG_WINDOW_HEIGHT |
+                XCB_CONFIG_WINDOW_BORDER_WIDTH);
```
Comment (ammen99) [inline src/view/xwayland.cpp]: why do we configure border width at all?

File: src/view/xwayland.cpp [hunk 3]
```diff
@@ -356,7 +359,7 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
         {
             last_server_width = xw->size_hints->base_width;
             last_server_height = xw->size_hints->base_height;
-            send_configure();
+            send_configure(XCB_CONFIG_WINDOW_X | XCB_CONFIG_WINDOW_Y);
```
Comment (ammen99) [inline src/view/xwayland.cpp]: huh, i think here we want to change width and height as well?

Comment (soreau) [reply src/view/xwayland.cpp]: Yes, it seems so.

File: src/view/xwayland.cpp (hunk 2 above)
Comment (soreau) [reply src/view/xwayland.cpp]: You're right, this shouldn't be needed.

File: src/view/xwayland.cpp (hunk 1 above)
Comment (soreau) [reply src/view/xwayland.cpp]: Without this, xterm ends up starting with 3x3 size.

**Comment by ammen99** [comment]
As discussed in IRC, this isn't necessary at all.

**Pull request closed without merging**